sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/FileSearchTestBase.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.assistants;
+
+import com.azure.ai.openai.assistants.models.AssistantCreationOptions;
+import com.azure.ai.openai.assistants.models.FileDetails;
+import com.azure.ai.openai.assistants.models.FileSearchToolDefinition;
+import com.azure.ai.openai.assistants.models.FileSearchToolDefinitionDetails;
+import com.azure.core.http.HttpClient;
+import com.azure.core.util.BinaryData;
+
+import java.util.Arrays;
+import java.util.function.BiConsumer;
+
+public abstract class FileSearchTestBase extends AssistantsClientTestBase {
```

Comment:
This is really cool. I think we could try grouping tests somehow with slightly more specific `TestBases` for the sake of setup and grouping of test resources for the suite.